react/ViewStack/index.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useContext } from 'react'
+import SwipeableViews from 'react-swipeable-views'
+
+// This is guessed, I could not see how to interface properly with
+// react-swipeable-views's onTransitionEnd
+const STEPPER_TRANSITION_DURATION_MS = 500
+
+export const ViewStackContext = React.createContext()
+export const useViewStack = () => useContext(ViewStackContext)
+
+const sleep = duration => new Promise(resolve => setTimeout(resolve, duration))
+
+const useStack = initialState => {
+  const [arr, setArray] = useState(initialState)
+  const [curIndex, setCurIndex] = useState(initialState.length - 1)
+
+  const push = item => {
+    const newArr = [...arr, item]
+    setArray(newArr)
+    setCurIndex(curIndex + 1)
+  }
+
+  const pop = async () => {
+    if (arr.length < 2) {
```

Comment:
Nit: I find `if (arr.length <= 1)` more descriptive in that case. At first I thought « why 2? ». It may be just me :sweat_smile:

Reply:
Done, I also thought it was weird.